<!-- Launch the irob dVRK intermediate robot controller-->

<launch>
	<group ns="saf">
	
	
  		<arg name="arm_typ" default="PSM2"/>
		<arg name="arm_name" default="arm_1"/>
		<arg name="camera_registration_file" 
		default="registration_psm2.yaml"/>
		
		<arg name="instrument_info_file" 
		default="prograsp_forceps.yaml"/>
	
			<include 
			file="$(find irob_robot)/config/dvrk_topic_names.xml" /> 
  	
			<node name="robot_server_$(arg arm_typ)" pkg="irob_robot" 
			type="robot_server_dvrk" output="screen">
  		
  				<param name="arm_typ" type="string" value="$(arg arm_typ)" />
				<param name="arm_name" type="string" value="$(arg arm_name)" />
				<!-- <param name="camera_registration_file" type="string" 	
									value="$(arg camera_registration_file)" /> -->
				<rosparam command="load" 
				file="$(find irob_robot)/config/$(arg camera_registration_file)"/>
				
				<rosparam command="load" 
				file="$(find irob_robot)/config/$(arg instrument_info_file)"/>
				
				
			</node>
  	</group>
</launch>
